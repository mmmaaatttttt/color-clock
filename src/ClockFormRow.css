.ClockFormRow.error .ClockFormRow--time,
.ClockFormRow.error p {
  color: red;
}

.ClockFormRow.error .ClockFormRow--time {
  border: 1px solid red;
  border-radius: 2px;
}

.ClockFormRow--heading {
  text-align: center;
  margin: 1.5rem 0 0.25rem;
}