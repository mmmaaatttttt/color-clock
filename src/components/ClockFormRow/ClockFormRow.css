.ClockFormRow.error .time {
  color: red;
  position: relative;
}

.ClockFormRow.error .time input {
  border: 1px solid red;
  border-radius: 2px;
  color: red;
}

.ClockFormRow.error .time span {
  color: red;
  font-size: 1.2rem;
  font-style: italic;
  font-weight: bold;
  position: absolute;
  bottom: -5px;
  right: 10px;
}

.ClockFormRow--heading {
  text-align: center;
  margin: 1.5rem 0 0.25rem;
}
