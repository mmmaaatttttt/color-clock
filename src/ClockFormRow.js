import React from "react";
import { capitalize } from "./utils";
import "./ClockFormRow.css";

function ClockFormRow({ prefix, values, handleChange, title, error = null }) {
  return (
    <div>
      <h4 className="ClockFormRow--heading">{title}</h4>
      <div className={`ClockFormRow row ${error ? "error" : ""}`}>
        <div className="column">
          <label htmlFor={`${prefix}Time`}>{capitalize(prefix)} Time</label>
          <input
            type="time"
            id={`${prefix}Time`}
            name={`${prefix}Time`}
            value={values[0]}
            onChange={handleChange}
            className="ClockFormRow--time"
          />
        </div>
        <div className="column">
          <label htmlFor={`${prefix}Color`}>{capitalize(prefix)} Color</label>
          <input
            type="color"
            id={`${prefix}Color`}
            name={`${prefix}Color`}
            value={values[1]}
            onChange={handleChange}
          />
        </div>
        {error && <p>{error}</p>}
      </div>
    </div>
  );
}

export default ClockFormRow;
