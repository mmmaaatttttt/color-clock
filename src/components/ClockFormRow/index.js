import React from "react";
import { capitalize } from "../../utils";
import "./ClockFormRow.css";

function ClockFormRow({
  prefix = "start",
  values = ["00:00", "#000000"],
  handleChange = () => {},
  title = "ClockFormRow",
  error = null
}) {
  return (
    <div>
      <h4 className="ClockFormRow--heading">{title}</h4>
      <div className={`ClockFormRow row ${error ? "error" : ""}`}>
        <div className="column time">
          <label htmlFor={`${prefix}Time`}>{capitalize(prefix)} Time</label>
          <input
            type="time"
            id={`${prefix}Time`}
            name={`${prefix}Time`}
            value={values[0]}
            onChange={handleChange}
          />
          {error && <span className="error-text">{error}</span>}
        </div>
        <div className="column">
          <label htmlFor={`${prefix}Color`}>{capitalize(prefix)} Color</label>
          <input
            type="color"
            id={`${prefix}Color`}
            name={`${prefix}Color`}
            value={values[1]}
            onChange={handleChange}
          />
        </div>
      </div>
    </div>
  );
}

export default ClockFormRow;
