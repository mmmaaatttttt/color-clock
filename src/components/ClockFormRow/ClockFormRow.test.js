import React from "react";
import { render } from "@testing-library/react";
import ClockFormRow from "./";

describe("basic ClockFormRow tests", function () {
  it("renders without crashing", function () {
    render(<ClockFormRow />);
  });

  it("matches snapshot", function () {
    const { asFragment } = render(<ClockFormRow />);
    expect(asFragment()).toMatchSnapshot();
  });
});
