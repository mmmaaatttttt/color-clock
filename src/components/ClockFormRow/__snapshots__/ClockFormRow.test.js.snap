// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic ClockFormRow tests matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h4
      class="ClockFormRow--heading"
    >
      ClockFormRow
    </h4>
    <div
      class="ClockFormRow row "
    >
      <div
        class="column time"
      >
        <label
          for="startTime"
        >
          Start Time
        </label>
        <input
          id="startTime"
          name="startTime"
          type="time"
          value="00:00"
        />
      </div>
      <div
        class="column"
      >
        <label
          for="startColor"
        >
          Start Color
        </label>
        <input
          id="startColor"
          name="startColor"
          type="color"
          value="#000000"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
